[IMP] ePoS: vulgarise the SSL ePos issue. (#3223)
odoo/documentation: content/applications/sales/point_of_sale/overview/epos_ssc.rst

ePos printers are explicitly designed to work with your Point of Sale system, which sends the
tickets directly to the printer.

To use the printer in Odoo, models that do not require an :doc:`IoT box
<../../../productivity/iot/config/connect>` may need to establish a secure connection between their
browser and the printer :doc:`with the HTTPS protocol <https>`. If so, trying to reach the printer
IP address in HTTPS induces a warning page.

.. figure:: epos_ssc/browser-https-insecure.png
   :align: center
   :alt: warning page about the connection privacy on Google Chrome

   Warning page on Google Chrome, Windows 10

In that case, you can force the connection by clicking :guilabel:`Advanced` and :guilabel:`Proceed
to [IP address] (unsafe)`. Doing so allows you to reach the page in HTTPS and use the ePOS printer
in Odoo as long as the browser window stays open.

.. warning::
   The connection is lost after closing the browser window. Therefore, this method should only be
   used as a **workaround** or as a pre-requisite for the following instructions.

For a long-term solution, generating, exporting, and importing a **self-signed certificate** is
mandatory.

.. note::

>    Please check the following list of compatible `Epson ePOS printers

Reviewer: I moved it because it didn't make sense to have it so early, but here it doesn't make sense either, so I'll probably move it again x-

   - TM-P60II (Peeler: Wi-Fi® model)
   - TM-P80 (Wi-Fi® model)

To use the printer in Odoo, models that do not require an :doc:`IoT box
<../../../productivity/iot/config/connect>` may need to establish a secure connection between their
browser and the printer with :doc:`the HTTPS protocol <https>`. If so, trying to reach the printer

> IP address in HTTPS induces a warning page.

Reviewer 2: ```suggestion
IP address in HTTPS leads to a warning page.
```
Although correct, "induces" sounds really weird and out-of-place in this context

.. figure:: epos_ssc/browser-https-insecure.png
   :align: center
   :alt: warning page about the connection privacy on Google Chrome

   Warning page on Google Chrome, Windows 10

In that case, you can force the connection by clicking :guilabel:`Advanced` and :guilabel:`Proceed
to [IP address] (unsafe)`. Doing so allows you to reach the page in HTTPS and use the ePOS printer
in Odoo as long as the browser window stays open.

.. warning::
   The connection is lost after closing the browser window. Therefore, this method should only be

>    used as a **workaround** or as a pre-requisite for the following instructions.

Reviewer 2: ```suggestion
   used as a **workaround** or as a pre-requisite for the following instructions.
```
Which following instructions ? Generate, export... ? Maybe use a :ref: here

                  #. restart the computer to make sure that the changes are applied.

               .. tab:: Linux

                  If you are using a **Google Chrome** browser,

                  #. open your Chrome browser;
                  #. go to :menuselection:`Settings --> Privacy and security --> Security -->
                     Manage certificates`;
                  #. go to the :guilabel:`Authorities` tab, click :guilabel:`Import`, and select
                     the exported certification file;
                  #. accept all warnings;
                  #. click :guilabel:`ok`; and
                  #. restart your browser.

                  If you are using a **Mozilla Firefox** browser,

                  #. open your Firefox browser;
                  #. go to :menuselection:`Settings --> Privacy & Security --> Security --> View
                     Certificates... --> Import`;
                  #. select the exported certification file;
                  #. tick the checkboxes and validate; and
                  #. restart your browser.

   .. tab:: Mac OS

      To secure the connection on a Mac:

>       #. open Safari and navigate to your printer's IP address. Doing so induces a warning page;

Reviewer 2: ```suggestion
      #. open Safari and navigate to your printer's IP address. Doing so displays a warning page;
```
Same as above, consider "leads to" or "displays"

      #. on the warning page,  go to :menuselection:`Show Details --> visit this website --> Visit

>          Website`, validate; and

Reviewer 2: ```suggestion
         Website`, validate; 
```

      #. reboot the printer so you can use it with any other browser.

   .. tab:: Android OS

      To import an SSL certificate into an Android device, first create and export it from a

>       computer. Next, transfer the .crt file to the device using email, Bluetooth, or USB. Once the

Reviewer 2: ```suggestion
      computer. Next, transfer the `.crt` file to the device using email, Bluetooth, or USB. Once the
```

      file is on the device,

>       #. open the settings and search for `certificate`.

Reviewer 2: ```suggestion
      #. open the settings and search for `certificate`;
```

      #. click :guilabel:`Certificate AC` (Install from device storage);
      #. select the certificate file to install it on the device.

      .. Note::
         The specific steps for installing a certificate may vary depending on the version of
         Android and the device manufacturer.

.. important::

>    - If you need to export SSL certificates from an operating system or web browser that hasn't been

Reviewer 2: ```suggestion
   - If you need to export SSL certificates from an operating system or web browser that has not been
```
Contractions are considered informal in written English

   - TM-P60II (Peeler: Wi-Fi® model)
   - TM-P80 (Wi-Fi® model)

To use the printer in Odoo, models that do not require an :doc:`IoT box
<../../../productivity/iot/config/connect>` may need to establish a secure connection between their
browser and the printer with :doc:`the HTTPS protocol <https>`. If so, trying to reach the printer

> IP address in HTTPS leads to a warning page.

Reviewer: I've mixed your suggestion with some alterations, tell me if that's clearer

            to its IP address, for example, `https://192.168.1.25`. Then, force the connection as
            explained in the :ref:`introduction <epos_ssc/ePOS printers>`.

            If you are using a **Google Chrome** browser,

            #. click :guilabel:`Not secure` next to the search bar, and :guilabel:`Certificate is
               not valid`;

               .. image:: epos_ssc/browser-warning.png
                  :align: center
                  :alt: Connection to the printer not secure button in Google Chrome browser.

            #. go to the :guilabel:`Details` tab and click :guilabel:`Export`;
            #. add `.crt` at the end of the file name to ensure it has the correct extension;
            #. select :guilabel:`Base64-encoded ASCII, single certificate`, at the bottom of the
               pop-up window;
            #. save, and the certificate is exported.

            .. warning::
               Make sure that the certificate ends with the extension `.crt`. Otherwise, some
               browsers might not see the file during the import process.

            If you are using a **Mozilla Firefox** browser,

>             #. click the **lock-shaped** icon next to the search bar;

Reviewer: You're right :)

                  certificates must be imported from the certification file rather than the browser.
                  To do so,

                  #. open the Windows File explorer and locate the downloaded certification file;
                  #. click right on the certification file and click :guilabel:`Install
                     Certificate`;
                  #. select where to install the certificate and for whom - either for the
                     :guilabel:`Current User` or all users (:guilabel:`Local Machine`). Then, click
                     :guilabel:`Next`;
                  #. on the `Certificate Store` screen, tick :guilabel:`Place all certificates in
                     the following store`, click :guilabel:`Browse...`, and select
                     :guilabel:`Trusted Root Certification Authorities`;

                     .. image:: epos_ssc/win-cert-wizard-store.png
                        :align: center

                  #. click :guilabel:`Finish`, accept the pop-up security window;
                  #. restart the computer to make sure that the changes are applied.

               .. tab:: Linux

>                   If you are using a **Google Chrome** browser,

Reviewer 3: ```suggestion
                  If you are using **Google Chrome**,
```

                  #. open your Chrome browser;
                  #. go to :menuselection:`Settings --> Privacy and security --> Security -->
                     Manage certificates`;
                  #. go to the :guilabel:`Authorities` tab, click :guilabel:`Import`, and select
                     the exported certification file;
                  #. accept all warnings;
                  #. click :guilabel:`ok`;
                  #. restart your browser.

>                   If you are using a **Mozilla Firefox** browser,

Reviewer 3: ```suggestion
                  If you are using **Mozilla Firefox** ,
```

                        Open your Chrome browser, go to :menuselection:`Settings --> Privacy and
                        security --> Security --> Manage certificates`

                        Go to the :guilabel:`Authorities` tab, click :guilabel:`Import` and select
                        the exported certification file. Accept all warnings, click :guilabel:`ok`
                        and restart your browser.

                     .. tab:: Mozilla Firefox

                        Open your Firefox browser, go to :menuselection:`Settings --> Privacy &
                        Security --> Security --> View Certificates... --> Import`, and select the
                        exported certification file. Tick the checkboxes and validate. Then, restart
                        your browser.

   .. tab:: Mac OS

      On Mac OS, open a Safari webpage to secure the connection. Then, navigate to your printer IP
      address. Doing so induces a warning page. From there, go to :menuselection:`Show Details -->
      visit this website --> Visit Website` and validate.

      Reboot the printer to use it from any other browser.

   .. tab:: Android OS

>       On your Android device, open the settings and search for *certificate*. Then, click

Reviewer: We could, but it'll need some testing because idk at the moment how to do it; I'll try today, and hopefully,y it works on the first try!
